util-core/src/main/scala/com/twitter/util/Future.scala

```diff
@@ -339,6 +339,33 @@ def join[%s](%s): Future[(%s)] = join(Seq(%s)) map { _ => (%s) }""".format(
   def join[A](fs: JList[Future[A]]): Future[Unit] = Futures.join(fs)
 
   /**
+   * Take a sequence of `() => Future[_]` and execute each
+   * future sequentially, then return all future results as a single Future[Seq[_]].
+   * 
+   * If during execution any of the futures throws an exception that exception will be returned and 
+   * the remaining futures will not be processed.
+   * 
+   * A use case for sequential processing is ordered operations: 
+   *   val orderedOperations =
+   *     Seq(
+   *       () => deleteItem(id),
+   *       () => undeleteItem(id),
+   *       () => deleteItem(id)
+   *     )
+   *   Future.sequence(orderedOperations)
+   * 
+   * @param fs a sequence of Function0s that returns Futures to be executed in sequence
+   * @return a `Future[Seq[A]]` containing the results of each future in fs
+   */
+  def sequence[A](fs: Seq[() => Future[A]]): Future[Seq[A]] = 
```

Comment:
I'm not a fan of calling this ```sequence``` because ```sequence``` is typically understood to transform an ```M[Future[A]]``` to a ```Future[M[A]]```.